pkg/cmd/opts/git.go

```diff
@@ -172,7 +172,20 @@ func (o *CommonOptions) GitServerHostURLKind(hostURL string) (string, error) {
 		return "", err
 	}
 
-	kind, err := kube.GetGitServiceKind(jxClient, kubeClient, devNs, hostURL)
+	var clusterAuthConfig *auth.AuthConfig
+
+	// If the jx-auth-config configmap exists, create its config and pass it to GetGitServiceKind
+	configMapClient := kubeClient.CoreV1().ConfigMaps(devNs)
```

Comment:
Why is the configmap check required here? 

We should use this logic for all cases:
```
clusterAuthConfigSvc, err := o.GitAuthConfigService()
		if err != nil {
			return "", errors.Wrapf(err, "getting cluster-based auth configmap for checking kind of git url %s", hostURL)
		}
		clusterAuthConfig = clusterAuthConfigSvc.Config()
```
The auth config service is meant to abstract the kubernetes check away. It does check for the existence of the configmap and creates the appropriate auth config service: See `pkg/cmd/clients/factory.go: CreateAuthConfigService`. When the secret is there, it will create the kuberntes auth service which parses the kind from the secret, so we don't need to perform this separately.

Reply:
Ok. I was just being super defensive here.